packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/useCreditCard.js

```
@@ -0,0 +1,439 @@
+                ({ message }) => {
+                    errors.push(message);
+                }
+            );
+        }
+
+        return errors;
+    }, [ccMutationErrors, billingAddressMutationErrors]);
+
+    /**
+     * Helpers
+     */
+
+    const setIsBillingAddressSameInCache = useCallback(() => {
+        client.writeQuery({
+            query: getIsBillingAddressSameQuery,
+            data: {
+                cart: {
+                    __typename: 'Cart',
+                    id: cartId,
+                    isBillingAddressSame
+                }
+            }
+        });
+    }, [client, cartId, getIsBillingAddressSameQuery, isBillingAddressSame]);
+
+    const setShippingAddressAsBillingAddress = useCallback(() => {
+        const shippingAddress = shippingAddressData
+            ? mapAddressData(shippingAddressData.cart.shippingAddresses[0])
+            : {};
+        const {
+            firstName = '',
+            lastName = '',
+            country = '',
+            street1 = '',
+            street2 = '',
+            city = '',
+            state = '',
+            postalCode = '',
+            phoneNumber = ''
+        } = shippingAddress;
+
+        updateBillingAddress({
+            variables: {
+                cartId,
+                firstName,
+                lastName,
+                street1,
+                street2,
+                city,
+                state,
+                postalCode,
+                country,
+                phoneNumber,
+                sameAsShipping: true
+            }
+        });
+    }, [updateBillingAddress, shippingAddressData, cartId]);
+
+    const setBillingAddress = useCallback(() => {
+        const {
+            firstName = '',
+            lastName = '',
+            country = '',
+            street1 = '',
+            street2 = '',
+            city = '',
+            state = '',
+            postalCode = '',
+            phoneNumber = ''
+        } = formState.values;
+
```

Wouldn't this end up suppressing any errors from GraphQL? Is that the behavior we want?

---

You are right, Ill change this.